### PR TITLE
Fix some compiler warnings

### PR DESCRIPTION
**Pull request by otaviobp**

### CODE REVIEW

File: src/samples/design_patterns/stream_sample.c [hunk 1]
```diff
@@ -245,7 +245,7 @@ struct my_stream_api_handle *
 my_stream_api_new(const struct my_stream_api_config *config, int dev)
 {
     struct my_stream_api_handle *handle;
-    size_t data_buffer_size;
+    size_t data_buffer_size = 0;
```
Comment (cabelitos) [inline src/samples/design_patterns/stream_sample.c]: I really don't know why gcc did not warn me about this one

File: src/modules/flow/process/output.c [hunk 2]
```diff
@@ -76,7 +76,7 @@ output_write(struct output_data *output)
         } else if (r < 0) {
             if (errno == EINTR)
                 continue;
-            else if (errno == EAGAIN || errno == EWOULDBLOCK)
+            else if (errno == EAGAIN)
```
Comment (cabelitos) [inline src/modules/flow/process/output.c]: Well, what are we going to do on those "non supported systems" if the values are different?

Comment (ibriano) [inline src/modules/flow/process/output.c]: On Thu, 19 May 2016 10:35:02 -0700, Guilherme Íscaro wrote:
> > @@ -76,7 +76,7 @@ output_write(struct output_data *output)
> >          } else if (r < 0) {
> >              if (errno == EINTR)
> >                  continue;
> > -            else if (errno == EAGAIN || errno == EWOULDBLOCK)
> > +            else if (errno == EAGAIN)
> 
> Well, what are we going to do on those "non supported systems" if the values are different?

Plan on porting Soletta for AIX?

Comment (cabelitos) [inline src/modules/flow/process/output.c]: You never know....

**Comment by otaviobp** [comment]
merged.

**Pull request closed without merging**